nvim/lsp/ruff.lua

```diff
@@ -0,0 +1,22 @@
+-- Disable hover in favor of Pyright
+vim.api.nvim_create_autocmd("LspAttach", {
+    group = vim.api.nvim_create_augroup("lsp_attach_disable_ruff_hover", { clear = true }),
+    callback = function(args)
+        local client = vim.lsp.get_client_by_id(args.data.client_id)
+        if client == nil then
+            return
+        end
+        if client.name == "ruff" then
+            client.server_capabilities.hoverProvider = false
```

Comment:
Note: this can be verified by opening a Python file and running
```
:lua =vim.lsp.get_active_clients()[2].server_capabilities
```
Since `pyright` and `ruff` are attached, one can further verify that the right one is inspected by printing `name` instead of `server_capabilities`.